share/completions/mkdir.fish

```diff
@@ -0,0 +1,9 @@
+complete -c mkdir -s m -l mode --description 'Set file mode (as in chmod)' -x
+complete -c mkdir -s p -l parents --description 'Make parent directories as needed'
+complete -c mkdir -s v -l verbose --description 'Print a message for each created directory'
+complete -c mkdir -s Z --description 'Set SELinux security context of each created directory to the default type'
```

Comment:
What has worked well for us is something like

```fish
if mkdir --version >/dev/null ^/dev/null
# GNU
else
# BSD/OSX
```

this will also work if you install GNU tools on OSX, and will also work on Hurd.

I think the only place we actually check `uname` for completions is killall, but that's a special case (you can't _ever_ invoke it on Solaris).